src/realm/parser/parser.cpp

```diff
@@ -28,6 +28,8 @@
 
 #include <realm/parser/parser_utils.hpp>
 
+// clang-format off
```

Comment:
Is this really necessary? In most cases I've found adding a `//` at the end of the like where I want a line break to be sufficient to get good formatting from clang-format.

Reply:
I think this section warrants an exception due to the odd templated structure of the pegtl parser syntax. In my opinion clang format would make the code unreadable. Happy to be dissuaded of this though.

Reply:
If we added `SplitEmptyRecord: false` it actually comes out pretty nicely. But since it makes a lot of whitespace changes to remove the spaces inside `<>`, it would be best to defer to a separate commit anyway.